nntrainer/models/neuralnet.h

```diff
@@ -251,10 +257,14 @@ class NeuralNetwork : public ml::train::Model {
   /**
    * @brief     Run NeuralNetwork train
    * @param[in] values hyper parameters
+   * @param[in] stop_cb callback function to decide stop training or not
```

Comment:
Please describe the semantics of 
1. stop_cb's return
2. stop_cb's input parameter
in the header function doxygen.

Reply:
According to metal's reply on https://stackoverflow.com/questions/54483956/how-to-document-a-function-that-takes-a-callback, Doxygen doesn't support document the functor itself in a nested fashion. So I workaround using tildes. Please let me know if there is a better way :)

Reply:
You can just describe it in normal human language. :)